### PR TITLE
Follow PTL docs for setting tensor devices

### PR DESCRIPTION
**Pull request by piraka9011**
Following the [official PTL docs](https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html#delete-cuda-or-to-calls) for multi-gpu/tpu support, remove all `.to()` and `.cuda()` calls and use `.type_as()` instead.

### CODE REVIEW

**Comment by SeanNaren** [comment]
Hey man! any reason you closed this?

**Comment by piraka9011** [comment]
By using `mask = mask.type_as(x)`, mask get's converted to a `HalfTensor` for some reason.

I think we can remove `inputs = inputs.to(self.device)` but not sure about the cuda block. Need to do a bit more testing here.

**Comment by SeanNaren** [comment]
We could probably do:

```
mask = torch.tensor(x.size(), dtype=torch.bool, device=x.device).fill_(0)
```
